### ISSUE TITLE
Wide Configuration (Number of FM)

### ISSUE THREAD

**Issue by samieti02** [Kamnitsask/deepmedic#170]
Hello my dear Sir,

why do you use so much FM per layer?
`numberFMsPerLayerNormal = [30, 40, 50, 60, 70, 80, 90, 100]`

I wonder, if you change it like this, you will not have also good results?
numberFMsPerLayerNormal = [15, 15, 15, 15, 15, 15, 15]

Let s say, you have 100 flair and 50 t2 images of size 200x200x200 with 2 classes, and want to train a model for all images. You add each of the files in trainConfig.cfg trainGtLabels.cfg trainRoiMasks.cfg. e.g. in one trainRoiMasks, you have flair and t2 files, meaning each file has its own roi and label file. And you have trainChannels_t2_flair.cfg, where you added all the files. 

What can be a good configuration for the model to get reasonable results?

Thanks 
Sami

**Comment by samieti02** [comment]
Hello again sir,

lets say you have a data set with 150 Images.
What would you recommand one? To use the set like this 50-30-20 or 60-20-20 (Train-Validate-Test)?

Why do yo use so many FM in layers [..., 90, 100] ? As i first asked , 15 FM per layers would not be sufficient ? If no, why?

Thanks Sir

**Comment by samieti02** [comment]
About the type of sampling, if you just have 2 classes, class 0 = non tumor or background and class 1 = tumor or foreground (or 2 classes for tumor and non tumor), there would not be a difference between `typeOfSamplingForTraining = 0` and `typeOfSamplingForTraining = 3`  or which type would be better in this case?
How can we process more the class1, by setting the value `proportionOfSamplesToExtractPerCategoryTraining = [0.3, 0.7]` like this, for type = 3 OR 
`proportionOfSamplesToExtractPerCategoryTraining = [0.7, 0.3]` like this, for type = 1 ?

Because I have Flair and T1 modalities and for Flair, i have a score of 70% and for T1 50%.  I try now differents settings. Is it also possible that you integrate the mean accuracy over all epochs for the training ?

Thanks
Sami

**Comment by Kamnitsask** [comment]
Hello Samieti02,

First, regarding the issues, please try to keep create separate github issues for different problems (number of FMs and type of sampling are unrelated)

There are no rule of thumb for number of FMs. It's empyrical. There is some theory on this, but this issue tracker is not for this, sorry. For any task, you make the model bigger while performance increases, you stop when performance decreases/overfits. Lets say it's trial and error.

The defaults that I provide are not optimal on "any" task. I am running multiple experiments on various databases over the course of my PhD, and I simply try to provide defaults that are as good as possible "on average", so that DeepMedic gives in general good performance out of the box for a random user/task. But I cannot guarantee optimality. Users have to take care of that.
